src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SemanticModelExtensions.cs

```diff
@@ -234,7 +237,11 @@ private static bool CanBindToken(SyntaxToken token)
                 else if (current.Parent is ForEachStatementSyntax foreachStatement &&
                          foreachStatement.Expression == expression)
                 {
-                    return foreachStatement.Identifier.ValueText.ToCamelCase().Pluralize();
+                    var name = foreachStatement.Identifier.ValueText.ToCamelCase();
+#if !CODE_STYLE
+                    name = name.Pluralize();
+#endif
+                    return name;
```

Comment:
i don't love this.  either this isn't used by codestyle, in which case i'd move the entire extension method to a higher layer.  or it is used, but now this doesn't work at that layer.

Reply:
I am fine to do it as a separate change. This is need to unblock the break to CodeStyle package missing a dependency.

Reply:
yeah this appears to only be used by the [GenerateConstructorService](https://github.com/dotnet/roslyn/blob/master/src/Features/CSharp/Portable/GenerateConstructor/CSharpGenerateConstructorService.cs). We should refactor this

---
In reply to: [480324097](https://github.com/dotnet/roslyn/pull/47297#discussion_r480324097) [](ancestors = 480324097)

Reply:
couldn't we just move it in this PR?  seems like it would be really simple and much cleaner.

Reply:
I am not very sure how exactly we would prefer to refactor this - 2 separate API overloads, one which is available in shared layer and never pluralizes + another which is in the Workspaces layer which does pluralize?

Reply:
if it's called from codestyle, i would want it to support this, in which case, have a reference to humanizer.

but it sounds like it's not called from codestyle.  in which case, we pull the entire method up to the layer where it is needed (Features, i think).